src/main/java/com/atlassian/db/replica/api/ThrottledCache.java

```diff
@@ -17,58 +18,76 @@
  */
 @ThreadSafe
 public final class ThrottledCache<T> implements SuppliedCache<T> {
-    private final AtomicReference<Instant> lock = new AtomicReference<>();
-    private final Clock clock;
-    private final Duration timeout;
-    private T value = null;
+    private final SuppliedCache<T> delegate;
 
     public ThrottledCache(Clock clock, Duration timeout) {
-        this.clock = clock;
-        this.timeout = timeout;
+        this.delegate = new ThrottledCacheWithTimeout<>(clock, timeout);
```

Comment:
ahh, this is how you hide it, and make it backward compatible. 
IMHO visibility of this change from a customer perspective is low.
Maybe java doc? `public ThrottledCache(Clock clock, Duration timeout) {`